packages/diagrams/frontend/sirius-components-diagrams/src/renderer/panel/DiagramPanelContribution.tsx

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (c) 2023, 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import ClickAwayListener from '@material-ui/core/ClickAwayListener';
+import IconButton from '@material-ui/core/IconButton';
+import Paper from '@material-ui/core/Paper';
+import Popper from '@material-ui/core/Popper';
+import Tooltip from '@material-ui/core/Tooltip';
+
+import { Selection, useSelection } from '@eclipse-sirius/sirius-components-core';
+import { useRef, useState } from 'react';
+import { DiagramPanelContributionProps } from './DiagramPanelContribution.types';
+
+export const DiagramPanelContribution = ({ view, editingContextId, diagramId }: DiagramPanelContributionProps) => {
+  const [state, setState] = useState<boolean>(false);
```

Comment:
Please set meaningful names for `state` and `setState`

Reply:
done